cli/server/server.go

```diff
@@ -530,12 +533,14 @@ Main:
 				log.Info("SIGHUP received, restarting rpc-server")
 				rpcServer.Shutdown()
 				rpcServer = server.New(chain, cfg.ApplicationConfiguration.RPC, serv, oracleSrv, log, errChan)
-				rpcServer.Start()
+				serv.AddService(&rpcServer) // Replaces old one by service name.
+				if !cfg.ApplicationConfiguration.RPC.StartWhenSynchronized || serv.IsInSync() {
+					rpcServer.Start()
+				}
```

Comment:
So it doesn't start automatically, only on `SIGHUP`?

Reply:
Either it's in the default configuration (StartWhenSynchronized false) and it works the way it worked before, or it's configured to StartWhenSynchronized and it only starts when the node is synchronized.